kotlin/src/main/kotlin/org/apache/fury/serializer/kotlin/UnsignedArraySerializers.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+@file:OptIn(ExperimentalUnsignedTypes::class)
+
+package org.apache.fury.serializer.kotlin
+
+import org.apache.fury.Fury
+import org.apache.fury.memory.MemoryBuffer
+import org.apache.fury.serializer.Serializer
+import org.apache.fury.type.Type
+
+abstract class AbstractDelegatingArraySerializer<T, T_Delegate>(
+    fury: Fury,
+    cls: Class<T>,
+    private val delegateClass: Class<T_Delegate>
+) : Serializer<T> (fury, cls) {
+
+    abstract fun toDelegateClass(value: T): T_Delegate
+
+    abstract fun fromDelegateClass(value: T_Delegate): T
+
+    override fun getXtypeId(): Short {
+        return (-Type.LIST.id).toShort()
+    }
+
+    override fun xwrite(buffer: MemoryBuffer, value: T) {
+        write(buffer, value)
+    }
+
+    override fun xread(buffer: MemoryBuffer): T {
+        return read(buffer)
+    }
+
+    override fun write(buffer: MemoryBuffer, value: T) {
+        val delegatingSerializer = fury.classResolver.getSerializer(delegateClass)
```

Comment:
Could we make this as a field of the DelegatingArraySerializer to reduce lookup cost?

Reply:
All those unsiged array type are final I guess

Reply:
The lazy can be removed too